devito/arguments.py

```diff
@@ -209,8 +206,23 @@ def dtype(self):
 
     @cached_property
     def rtargs(self):
-        size = ScalarArgument("%s_size" % self.name, self, max)
-        return [size]
+        size = ScalarArgument(self.size_name, self, max)
+        start = ScalarArgument(self.start_name, self, max, 0)
+        end = ScalarArgument(self.end_name, self, max)
+        return [size, start, end]
+
+    def promote(self, value):
+        if not isinstance(value, tuple):
+            size, start, end = self.rtargs
+            value = (value, start.default_value, value)
+        else:
+            if len(value) == 2:
+                # Assume we've been passed a (start, end) tuple
```

Comment:
is "Assume" the right word ? isn't it always the case if we have a 2-tuple ?

Reply:
There are many ways to interpret a 2-tuple in order to promote it to a 3-tuple. I just chose one as convention and stated that assumption here. It is an assumption.